lib/src/utils/w3c_header_utils.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:instabug_flutter/src/models/generated_w3_header.dart';
+
+class W3CHeaderUtils {
+  static Random _random = Random();
+
+  W3CHeaderUtils();
+
+  @visibleForTesting
+  // ignore: use_setters_to_change_properties
+  static void $setRandom(Random random) {
+    _random = random;
+  }
+
+  /// Generate random 32 bit unsigned integer Hexadecimal (8 chars) lower case letters
+  /// Should not return all zeros
+  static Map<String, dynamic> generateTracePartialId() {
+    int randomNumber;
+    String hexString;
+
+    do {
+      randomNumber = _random.nextInt(0xffffffff);
+      hexString = randomNumber.toRadixString(16).padLeft(8, '0');
+    } while (hexString == '00000000');
+
+    return {
+      'numberPartialId': randomNumber,
+      'hexStringPartialId': hexString.toLowerCase(),
+    };
```

Comment:
Why not turn that into a simple class as well for type-safety? I don't think using maps for structured data is ideal.

Besides, what do we need the `numberPartialId` for?